### ISSUE TITLE
Large tests run out of memory

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/dtslint#129]
For very large tests, such as for Lodash, dtslint eventually runs out of memory.

Is there any way to workaround this issue?

Currently, people trying to contribute to the Lodash typings (and probably others) are being faced with sporadic memory errors, when trying to run the tests locally and in the Travis build. The [advice](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a5ab4665a849d139572aeb89a30f739c12b3bdcf/types/lodash/readme.md) is to try and rewrite the tests to avoid tipping the tests over the line of failure, but this is not really practice.

**Comment by andy-ms** [comment]
Out of curiousity, I measured memory usage on various ts versions:
```ts
| Version | Memory used |
| --- | --- |
| 2.2 | 527878K |
| 2.3.4 | 516141K |
| 2.4.2 | 877871K |
| 2.5.3 | 432805K |
| 2.6.2 | 230015K |
| 2.7.2 | 895674K |
| 2.8.4 | 242078K |
| 2.9.2 | 240159K |
| 3.1.0-dev.20180717 | 248034K |
```

dtslint tests against the lowest and highest versions supported. The lowest is currently 2.2 which used 530M. We might consider upgrading to 2.6 which used only 230M. (We'll have to remember to skip over 2.7 which used 900M!)
@rbuckton and @mhegazy were recently considering doing a broad upgrade to versions on DefinitelyTyped.

**Comment by mhegazy** [comment]
We have taked about moving to 2.2 as the baseline in the past. 2.7 issue seems a bit problematic since it is smack dap in the middle.

**Comment by OliverJAsh** [comment]
Would it help if the tests were broken into separate files? This would also improve the editing experience—currently my IDE is very slow when working on the Lodash typing tests.

If so, is this something dtslint/DefiniteltyTyped would need to support?

**Comment by andy-ms** [comment]
dtslint already supports multiple files, but I don't think that would change anything, since it needs the whole program (with all the files) to work.

**Comment by andy-ms** [comment]
@mhegazy what about upgrading just lodash? It looks like almost all of its dependers have `lodash` in the name. (Exceptions: #27450, #27451, some others that re-expose lodash)

**Comment by mhegazy** [comment]
> what about upgrading just lodash?

what version are you thinking about? it is now at 2.2, do you want to go all the way to 2.8?

**Comment by mhegazy** [comment]
2.6 was released in Nov 2017. So that is 8 months ago. i would think it is fair to upgrade.

**Comment by andy-ms** [comment]
Should be fixed by DefinitelyTyped/DefinitelyTyped#27510.
